packages/zambdas/src/ehr/get-lab-orders/helpers.ts

```
@@ -0,0 +1,754 @@
+    },
+    {
+      name: 'code:missing',
+      value: 'false',
+    },
+    {
+      name: '_include',
+      value: 'ServiceRequest:encounter',
+    },
+
+    // it's only PST tasks, because RFRT and RPRT tasks are based on DiagnosticReport, they should be requested later
+    {
+      name: '_revinclude',
+      value: 'Task:based-on',
+    },
+    {
+      name: '_revinclude',
+      value: 'DiagnosticReport:based-on',
+    },
+
+    // include Observations
+    {
+      name: '_include:iterate',
+      value: 'DiagnosticReport:result',
+    },
+  ];
+
+  if (encounterId) {
+    searchParams.push({
+      name: 'encounter',
+      value: `Encounter/${encounterId}`,
+    });
+  }
+
+  if (patientId) {
+    searchParams.push({
+      name: 'subject',
+      value: `Patient/${patientId}`,
+    });
+  }
+
+  if (serviceRequestId) {
+    searchParams.push({
+      name: '_id',
+      value: serviceRequestId,
+    });
+  }
+
+  if (visitDate) {
+    searchParams.push({
+      name: 'authored',
+      value: visitDate,
+    });
+  }
+
+  return searchParams;
+};
+
+export const extractLabResources = (
+  resources: (ServiceRequest | Task | DiagnosticReport | Encounter | Observation)[]
+): {
+  serviceRequests: ServiceRequest[];
+  tasks: Task[];
+  diagnosticReports: DiagnosticReport[];
+  encounters: Encounter[];
+  observations: Observation[];
+} => {
+  const serviceRequests: ServiceRequest[] = [];
+  const tasks: Task[] = [];
+  const diagnosticReports: DiagnosticReport[] = [];
+  const encounters: Encounter[] = [];
+  const observations: Observation[] = [];
+  for (const resource of resources) {
+    if (resource.resourceType === 'ServiceRequest') {
+      const serviceRequest = resource as ServiceRequest;
+      const withActivityDefinition = serviceRequest.contained?.some(
+        (contained) => contained.resourceType === 'ActivityDefinition'
+      );
+      if (withActivityDefinition) {
+        serviceRequests.push(serviceRequest);
+      }
```

If a ServiceRequest didn't have an ActivityDefinition, i.e., no Orderable Item info, that would definitely be an error case. Do we want to surface that error here? Or just let Oystehr submit-lab handle that validation?

---

I think this is ultimately being used to populate the lab order table, so i don't think we would want to error and cause the whole table to not load if one of the service requests is bunked. However, I think if the activity definition is missing some critical info will be missing be tables and possibly the order form. It would probably make sense to error when generating the order pdf if this is missing.